src/middleware.rs

```diff
@@ -384,9 +428,128 @@ pub enum OperationArg {
     Key(AnchoredKey),
 }
 
+impl OperationArg {
+    pub fn is_none(&self) -> bool {
+        matches!(self, Self::None)
+    }
+    pub fn statement(&self) -> Result<Statement> {
+        match self {
+            Self::Statement(statement) => Ok(statement.clone()),
+            _ => Err(anyhow!("Operation argument {:?} is not a statement.", self)),
+        }
+    }
+    pub fn key(&self) -> Result<AnchoredKey> {
+        match self {
+            Self::Key(ak) => Ok(ak.clone()),
+            _ => Err(anyhow!("Operation argument {:?} is not a key.", self)),
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Operation(pub NativeOperation, pub Vec<OperationArg>);
 
+impl Operation {
+    pub fn code(&self) -> NativeOperation {
+        self.0
+    }
+    pub fn args(&self) -> &[OperationArg] {
+        &self.1
+    }
+    // TODO: Argument checking.
+    /// Checks the given operation against a statement.
+    pub fn check(&self, output_statement: Statement) -> Result<bool> {
```

Comment:
About the return type:
Is the `Err` case reserved for malformed operation/statements, and the `Ok(false)` for properly formed data but invalid deduction?